### DISCUSSION TITLE
Allow `id` inputs to be `string` or `number` if the id type is `Int`

### DISCUSSION THREAD

**Discussion by flybayer** [prisma/prisma#2575] [Q&A] [answered]
We have an unpleasant DX experience with Blitz/Next.js because Prisma strictly enforces the id input type. 

We have routes like `/pages/[id]` which results in a params object of `{id: '123'}` for a route of `/pages/123`.

So this means that in every single component where we are using the route parameter, we have to manually run `parseInt(id)` before passing to Prisma.

It would be super nice if Prisma would also allow string inputs for ids and automatically handle `parseInt`. 

Rails does this, and it's very nice because you never have to think about it.

**Comment by ryands17** [comment, marked as answer]
Hey @flybayer :wave: 
Could you create a Feature Request [here](https://github.com/prisma/prisma/issues/new?assignees=&labels=&template=feature_request.md&title=) for the same?

**Reply by flybayer** [reply]
Done: https://github.com/prisma/prisma/issues/2580

**Comment by ludralph** [comment]
Hi @flybayer,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
